### PR TITLE
INFRA-1312 - Add ability to query private organizations

### PR DESCRIPTION
**Pull request by dhruvsingh**

### CODE REVIEW

**Review by Kaniabi — APPROVED** [review]

File: index.js [hunk 1]
```diff
@@ -41,21 +42,22 @@ async function auth(user, password, config) {
...
-  const {login, organizations_url} = res.body; // eslint-disable-line camelcase
+  const auth_api = `${API_URL}${USER_API}`; // eslint-disable-line camelcase
+  const res = await got.get(auth_api, options);
+  const {login} = res.body; // eslint-disable-line camelcase
 
   if (login.toLowerCase() !== user) {
     throw new Error('Invalid user');
   }
 
+  const organizations_url = `${API_URL}orgs/${config.org}/members/${user}`; // eslint-disable-line camelcase
```
Comment (4fuss) [inline index.js:53]: Feels a little bit messy with the slashes here and there. Let's tide it up.

File: index.js [hunk 2]
```diff
@@ -41,21 +42,22 @@ async function auth(user, password, config) {
...
-  const orgs = res2.body.map(org => org.login);
 
-  if (!orgs.some(org => org === config.org)) {
-    throw new Error(`User ${user} is not a member of ${config.org}`);
+  if (res2.statusCode !== 204) {
+    throw new Error(`User ${user} is not a member of ${config.org}. Error ${res2.body}`);
   }
 
-  return orgs;
+  return [config.org];
```
Comment (4fuss) [inline index.js:60]: Can we have a doctring for this function and what is expected to be returned?

**Review by 4fuss — CHANGES_REQUESTED** [review]

File: index.js (hunk 1 above)
Comment (dhruvsingh) [reply index.js:53]: Marking this resolved per slack conversation

File: index.js (hunk 2 above)
Comment (dhruvsingh) [reply index.js:60]: sure

**Review by utek — APPROVED** [review]

File: readme.md [hunk 3]
```diff
@@ -45,7 +42,3 @@ See [got#timeout](https://www.npmjs.com/package/got#timeout).
 * `httpRetries: number, default=2`
 
 See [got#retries](https://www.npmjs.com/package/got#retries).
-
-## License
-
-MIT, 2018 (c) Dmitriy Tsvettsikh
```
Comment (utek) [inline readme.md:51]: Should we remove this part?

**Review by utek — CHANGES_REQUESTED** [review]

**Review by Backscratcher — APPROVED** [review]

File: readme.md (hunk 3 above)
Comment (dhruvsingh) [reply readme.md:51]: yes

**Review by 4fuss — APPROVED** [review]

**Review by utek — APPROVED** [review]

File: readme.md (hunk 3 above)
Comment (dhruvsingh) [reply readme.md:51]: Sorry, Mis-understood the comment, will fix it

**Pull request merged**